src/qt/bitcoingui.cpp

```diff
@@ -1223,6 +1235,22 @@ void BitcoinGUI::resetblockchainClicked()
     }
 }
 
+bool BitcoinGUI::tryQuit()
+{
+    if(clientModel->getOptionsModel()->getConfirmOnClose() &&
```

Comment:
Since this is also directly invoked from the systray it might be wise to double check that the client- and options model aren't null pointers.